aziotctl/aziotctl-common/src/system/status.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::fmt;
+use std::process::Command;
+
+use anyhow::{Context, Result};
+
+use super::ServiceDefinition;
+
+pub fn get_status(processes: &[&ServiceDefinition]) -> Result<()> {
+    let services: Vec<ServiceStatus<'_>> = processes
+        .iter()
+        .map(|process| -> Result<ServiceStatus<'_>> {
+            Ok(ServiceStatus {
+                name: process.service,
+                state: State::from_systemctl(process.service)?,
+                sockets: process
+                    .sockets
+                    .iter()
+                    .map(|socket| -> Result<SocketStatus<'_>> {
+                        Ok(SocketStatus {
+                            name: socket,
+                            state: State::from_systemctl(socket)?,
+                        })
+                    })
+                    .collect::<Result<Vec<SocketStatus<'_>>>>()?,
+            })
+        })
+        .collect::<Result<Vec<ServiceStatus<'_>>>>()?;
+
+    println!("System services:");
+    for service in &services {
+        println!(
+            "    {:24}{}",
+            format!(
+                "{}:",
```

Comment:
minor ask... Can we drop the semi-colon ':' after the service name?  I know it was in the text I pasted into the bug description.  I forgot to remove it which is my bad.

Reply:
done